evalml/pipelines/components/transformers/samplers/undersampler.py

```diff
@@ -0,0 +1,93 @@
+
+        Arguments:
+            balanced_ratio (float): The largest majority:minority ratio that is accepted as 'balanced'. For instance, a 4:1 ratio would be
+                represented as 4, while a 6:5 ratio is 1.2. Must be greater than or equal to 1 (or 1:1). Defaults to 4.
+            min_samples (int): The minimum number of samples that we must have for any class, pre or post sampling. If a class must be downsampled, it will not be downsampled past this value.
+                To determine severe imbalance, the minority class must occur less often than this and must have a class ratio below min_percentage.
+                Must be greater than 0. Defaults to 100.
+            min_percentage (float): The minimum percentage of the minimum class to total dataset that we tolerate, as long as it is above min_samples.
+                If min_percentage and min_samples are not met, treat this as severely imbalanced, and we will not resample the data.
+                Must be between 0 and 0.5, inclusive. Defaults to 0.1.
+            random_seed (int): The seed to use for random sampling. Defaults to 0.
+        """
+        parameters = {"balanced_ratio": balanced_ratio,
+                      "min_samples": min_samples,
+                      "min_percentage": min_percentage}
+        parameters.update(kwargs)
+        sampler = BalancedClassificationSampler(**parameters, random_seed=random_seed)
+
+        super().__init__(parameters=parameters,
+                         component_obj=sampler,
+                         random_seed=random_seed)
+
+    def fit(self, X, y):
+        """Resample the data using the undersampler. Since our sampler doesn't need to be fit, we do nothing here.
+
+        Arguments:
+            X (ww.DataFrame): Training features
+            y (ww.DataColumn): Target features
+
+        Returns:
+            self
+        """
+        if y is None:
+            raise ValueError("y cannot be none")
+        return self
+
+    def fit_transform(self, X, y):
+        """Fit and transform the data using the undersampler. Used during training of the pipeline
+
+        Arguments:
+            X (ww.DataFrame): Training features
+            y (ww.DataColumn): Target features
+
+         Returns:
+            ww.DataTable, ww.DataColumn: Undersampled X and y data
+        """
+        X = infer_feature_types(X)
+        if y is None:
+            raise ValueError("y cannot be none")
+        y = infer_feature_types(y)
+        X_pd = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y_pd = _convert_woodwork_types_wrapper(y.to_series())
+        index_df = pd.Series(y_pd.index)
+        indices = self._component_obj.fit_resample(X_pd, y_pd)
+        train_indices = index_df[index_df.isin(indices)].dropna().index.values.tolist()
+        return X.iloc[train_indices], y.iloc[train_indices]
+
+    def transform(self, X, y):
```

Comment:
In our base `Transformer` class, we end up trying to [transform with self._component_obj](https://github.com/alteryx/evalml/blob/cdb769950586768ee8b012f1d7fb4254edb6f379/evalml/pipelines/components/transformers/transformer.py#L44), which would raise an error since we don't have a `transform` method with the undersampler. I overrode it to avoid this issue.

Reply:
Hmmm, this is curious.  Does it make more sense to raise a NotImplementedError?

Reply:
Hmm, I don't think so. When we try to score/predict on a pipeline, we call `transform()` on all of the transformer components rather than `fit_transform`, which makes sense since we don't want to refit the component if we have already trained the pipeline and are predicting. In this scenario, our sampler shouldn't do anything (since we only sample during training), and so we should allow users to both call `transform()` and get a result from it.

Reply:
I think this is a cool way of specifying "train only" transformers! I was thinking we'd have to pass info from the pipeline down to the component but I think it's great we can do something simpler.

The one confusing thing is that `fit().transform() != fit_transform()` which I think is what someone coming fresh off sklearn would expect. Maybe we should just be super clear about which transformers are "train only" and what users should expect with them. I think "train only" transformers are a cool concept we can use to differentiate ourselves.

Reply:
Yeah, that's a good point! Would it be best to state that this is a 'train only' transformer in the documentation/description of the component?